README.md

```diff
@@ -24,21 +23,129 @@ Join us to help define the direction and implementation of this project!
+Kubernetes allows decoupling policy decisions from the inner workings of the API Server by means of [admission controller webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/), which are executed whenever a resource is created, updated or deleted. Gatekeeper is a validating (mutating TBA) webhook that enforces CRD-based policies executed by [Open Policy Agent](https://github.com/open-policy-agent/opa), a policy engine for Cloud Native environments hosted by CNCF as a sandbox-level project.
 
-Kubernetes compliance is enforced at the “runtime” via tools such as network policy and pod security policy. [gatekeeper](https://github.com/Azure/gatekeeper) extends the compliance enforcement at “create” event not at “run“ event. For example, a kubernetes service could answer questions like :
-
-* Can we whitelist / blacklist registries.
-* Not allow conflicting hosts for ingresses.
-* Label objects based on a user from a department.
-
-In addition to the `admission` scenario  it helps answer the `audit` question such as:
-
-* What are the policies that my cluster is violating.
-
-In the `authorization` scenario it's possible to block things like `kubectl get`, `kubectl port-forward` or even non-resource requests (all authorized request can be blocked).
+In addition to the `admission` scenario, Gatekeeper's audit functionality allows administrators to see what resources are currently violating any given policy.
 
+Finally, Gatekeeper's engine is designed to be portable, allowing administrators to detect and reject non-compliant commits to an infrastructure-as-code system's source-of-truth, further strenthening compliance efforts and preventing bad state from slowing down the organization.
 
+## Installation Instructions
 
+WARNING: It is not recommended to install Gatekeeper on a production cluster. The project is in alpha and may or may not uninstall cleanly.
+
+### Installation
+
+Currently the most reliable way of installing Gatekeeper is to build and install from HEAD:
+
+   * Make sure [Kubebuilder is installed](https://book.kubebuilder.io/getting_started/installation_and_setup.html)
+   * Clone the Gatekeeper repo to your local system
+   * Make sure you have a container registry you can write to that is readable by the target cluster
+   * cd to the repository directory
+   * run `make docker-build REPOSITORY=<YOUR DESIRED DESTINATION DOCKER IMAGE>`
+   * run `make docker-push-release REPOSITORY=<YOUR DESIRED DESTINATION DOCKER IMAGE>`
+   * make sure your kubectl context is set to the desired installation cluster
+   * run `make deploy`
+
+### Uninstallation
+
+Note that this is not a clean uninstall. There may be some CRDs that are not cleaned up
+
+   * cd to the repository directory
+   * run `make uninstall`
+
+## How to Use Gatekeeper
+
```

Comment:
Would it make sense to link to https://github.com/open-policy-agent/frameworks/tree/master/constraint I found that readme very helpful.

Reply:
adding link w/next push